handlers/handler.go

```diff
@@ -78,14 +80,21 @@ func IndexS3Object(s3objectURL string) {
 	//     <dataguid>/<uuid>/<filename>
 	//
 	// we want to keep the `<dataguid>/<uuid>` part
+	key = strings.Trim(key, "/")
 	split_key := strings.Split(key, "/")
 	var uuid string
-	if len(split_key) == 2 {
+	var filename string
+	foundGUID, errGUID := id.Parse(split_key[0])
+	foundDG, errDG := id.Parse(split_key[1])
+	if errGUID == nil && len(foundGUID) > 0 {
 		uuid = split_key[0]
+		filename = split_key[len(split_key)-1]
+	} else if errDG == nil && len(foundDG) > 0 {
```

Comment:
we need some unit test cases:
- for this feature, if "--include-subdirname"
- for this feature, if not "--include-subdirname"
- for the bug we found related to prefixes (see #31 #32)

Reply:
Yeah I agree that we should have unit tests for this. My recommendation would be to pull the logic that parses out `uuid` and `filename` into its own function to make it easier to test.

Once we have the logic isolated and testable, it will also be simpler to test all existing prefixes currently in use.

Reply:
Totally agree with this John. I will update the PR accordingly.